app/helpers/application_helper/button/middleware_domain_server_action.rb

```diff
@@ -1,12 +1,5 @@
 class ApplicationHelper::Button::MiddlewareDomainServerAction < ApplicationHelper::Button::MiddlewareServerAction
-
   def visible?
-    in_domain? && super
-  end
-
-  private
-
-  def in_domain?
-    @record.present? && (@record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?))
+    @record.try(:in_domain?) && super
```

Comment:
There was an error and I fixed it already. `#in_domain?` is defined on `MiddlewareServer`, and delegated to MS on `MiddlewareDeployment`. So there's no need to do this check, we just call the method and let the models decide how to respond.